chainstate/tx-verifier/src/transaction_verifier/accounts_undo_cache.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{btree_map::Entry, BTreeMap};
+
+use super::{
+    error::ConnectTransactionError, storage::TransactionVerifierStorageError, TransactionSource,
+};
+use common::{
+    chain::{Block, Transaction},
+    primitives::Id,
+};
+use pos_accounting::{
+    AccountingBlockUndo, AccountingBlockUndoError, AccountingTxUndo, PoSAccountingUndo,
+};
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct AccountsBlockUndo {
+    tx_undos: BTreeMap<Id<Transaction>, PoSAccountingUndo>,
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct AccountsBlockUndoEntry {
```

Comment:
Some times accounts... some times accounting... can we stick to Accounting everywhere?